utils/qzip.c

```diff
@@ -469,9 +469,8 @@ int makeOutName(const char *in_name, const char *out_name,
  * parent directory. */
 void mkPath(char *path, const char *dirpath, char *file)
 {
-    if (strlen(dirpath) + strlen(file) + 1 < MAX_PATH_LEN) {
-        snprintf(path, MAX_PATH_LEN, "%s/%s", dirpath, file);
-    } else {
+    if (snprintf(path, MAX_PATH_LEN, "%s/%s", dirpath, file) >= MAX_PATH_LEN) {
```

Comment:
I think this is a good change, but why not check the return value for negative values (errors) while at it? Or is is so that `snprintf` can only return an error when the format string is malformed? (I'm also not the repository owner.)

Reply:
I think that, ignoring ancient C libraries predating C standardization, and for this particular format string, a negative return value should only occur for logical errors such as `MAX_PATH_LEN == 0`. (Of course, a malformed format string would also do it.) It’s possible there are some exotic scenarios I am not thinking of (I/O error on locale data?).

I agree that checking for an error, and not just for truncation, would be better, even if it is very unlikely to occur in practice.

If it were my code, I would want to handle an over-long path more gracefully than by aborting, and I would not use `assert()` for checks I wanted to happen at runtime because—although the QATzip build system doesn’t appear to ever set `NDEBUG`—it’s too easy for `-DNDEBUG` to be added to the compiler flags and disable all the `assert`s. However, I’m aiming for a small “drive-by” contribution here.

Reply:
Anyway, I’ll update with this change, which implements your suggestion:

```diff
diff --git a/utils/qzip.c b/utils/qzip.c
index 1b8319e..8800cc0 100644
--- a/utils/qzip.c
+++ b/utils/qzip.c
@@ -469,8 +469,9 @@ int makeOutName(const char *in_name, const char *out_name,
  * parent directory. */
 void mkPath(char *path, const char *dirpath, char *file)
 {
-    if (snprintf(path, MAX_PATH_LEN, "%s/%s", dirpath, file) >= MAX_PATH_LEN) {
-        /* truncation occurred */
+    const int nprinted = snprintf(path, MAX_PATH_LEN, "%s/%s", dirpath, file);
+    if (nprinted >= MAX_PATH_LEN || nprinted < 0) {
+       /* truncated, or output error */
         assert(0);
     }
 }
```